pikaur/core.py

```diff
@@ -144,12 +144,12 @@ def isolate_root_cmd(cmd: List[str], cwd=None) -> List[str]:
     ]
     if cwd is not None:
         base_root_isolator += ['-p', 'WorkingDirectory=' + cwd]
-    if "http_proxy" in os.environ:
-        base_root_isolator += ['-E', 'http_proxy=' + os.environ.get('http_proxy')]
-    if "https_proxy" in os.environ:
-        base_root_isolator += ['-E', 'https_proxy=' + os.environ.get('https_proxy')]
-    if "ftp_proxy" in os.environ:
-        base_root_isolator += ['-E', 'ftp_proxy=' + os.environ.get('ftp_proxy')]
+    if "http_proxy" in os.environ and os.environ.get('http_proxy') is not None:
+        base_root_isolator += ['-E', 'http_proxy=' + str(os.environ.get('http_proxy'))]
```

Comment:
just `os.environ['http_proxy']` should be fine

Reply:
I agree, it should according to the docs, however they seem to be not quite right, the checks don't work as expected . Without using str() MyPy failed over here when the variables where not set:

```
pikaur/core.py:148: error: Unsupported operand types for + ("str" and "None")
pikaur/core.py:148: note: Right operand is of type "Optional[str]"
```

Reply:
i don't think what the warning which you showing is generated for `os.environ['http_proxy']` line